nsqd/http.go

```diff
@@ -494,7 +494,24 @@ func (s *httpServer) doStats(w http.ResponseWriter, req *http.Request, ps httpro
 	channelName, _ := reqParams.Get("channel")
 	jsonFormat := formatString == "json"
 
-	stats := s.ctx.nsqd.GetStats()
+	var (
+		state TopicStats
+		stats []TopicStats
+	)
+	if topicName == "" {
+		stats = s.ctx.nsqd.GetStats()
+	} else if channelName == "" {
+		state, err = s.ctx.nsqd.GetTopicStats(topicName)
+	} else {
+		state, err = s.ctx.nsqd.GetChannelStats(topicName, channelName)
+	}
+	if nil != err {
+		s.ctx.nsqd.logf(LOG_ERROR, "failed to get stats - %s", err)
+		return nil, http_api.Err{404, "topic/channel is not found"}
+	} else if nil == stats {
```

Comment:
This doesn't need to be an `else if`, because the block above returns. Also, same minor note applies here, please move the variable to the _left_ side of the comparison.